src/Traits/FactoryTrait.php

```diff
@@ -154,24 +155,27 @@ public static function create(
         ?int $hour = null,
         ?int $minute = null,
         ?int $second = null,
+        ?int $microsecond = null,
         $tz = null
     ): ChronosInterface {
         $year = $year ?? (int)date('Y');
-        $month = $month ?? date('n');
-        $day = $day ?? date('j');
+        $month = $month ?? date('m');
+        $day = $day ?? date('d');
 
         if ($hour === null) {
-            $hour = date('G');
+            $hour = date('H');
             $minute = $minute ?? date('i');
             $second = $second ?? date('s');
+            $microsecond = $microsecond ?? date('u');
```

Comment:
All these parts should use same `$now` timestamp.

It could be `2:59:59.999999` while taking hour, but then `3:00:00.000000` while taking minute/second/millisecond, and final time might be 2:00:00.000000`.